### ISSUE TITLE
Add allele counting function for individual calls

### ISSUE THREAD

**Issue by eric-czech** [pystatgen/sgkit#85]
We need a function to count alleles for individual calls in addition to one counting alleles across samples.  I propose we call it `count_call_alleles` and rename `count_alleles` to `count_variant_alleles`.

This would be akin to [GenotypeAlleleCountsArray](https://scikit-allel.readthedocs.io/en/stable/model/ndarray.html#genotypeallelecountsarray) and [AlleleCountsArray](https://scikit-allel.readthedocs.io/en/stable/model/ndarray.html#allelecountsarray) in scikit-allele.

**Comment by eric-czech** [comment]
An important concern with this is whether or not multiple alleles should be counted separately.  

If this function was to generate arrays of shape `(n_variant, n_sample)`, then it clearly implies that some reduction was performed across the known alleles.  The most obvious reduction is to count any non-reference allele, but this can drop important information for downstream tasks.  It could instead return an array of shape `(n_variant, n_sample, n_allele)` as @timothymillar mentions in https://github.com/pystatgen/sgkit/issues/21#issuecomment-671119651, and we should start with this assumption.

Implementing an efficient function that returns counts as an array of shape `(n_variants, n_samples, n_alleles)` with values equal to the number of times each allele appears across chromosomes is not trivial and should likely involve something like a `bincount` across the ploidy dimension.  The "I only care about non-reference allele counts" case on the other hand is much closer to trivial, e.g. `call_allele_counts = (ds['call_genotype'] > 0).sum(dim='ploidy')` or if a user didn't want to count partially missing calls then `xr.where(ds['call_genotype_mask'], -1, (ds['call_genotype'] > 0).sum(dim='ploidy'))`.  We could and probably should wrap this functionality in the API with some docs, but it's also a degenerate case from the `(n_variants, n_samples, n_alleles)` result where you could do the same thing with `call_allele_counts[..., 1:].sum(dim='alleles')` (to get non-reference allele counts).  This would not be as efficient though.

**tl;dr** This function should return a count array of shape `(n_variants, n_samples, n_alleles)` and missing values should be excluded from the counts.

**Comment by tomwhite** [comment]
Fixed in #114
